notification-server/src/bin/server.rs

```diff
@@ -0,0 +1,62 @@
+use axum::{
+    extract::{Json, Path, State},
+    http::StatusCode,
+    routing::put,
+    Router,
+};
+use clap::Parser;
+use dotenv::dotenv;
+use std::sync::Arc;
+use tokio_postgres::Config;
+use tracing::info;
+
+#[derive(Debug, Parser)]
+struct Args {
+    #[arg(
+        long = "db-connection",
+        help = "A connection string detailing the connection to the database used by the \
+                application.",
+        env = "NOTIFICATION_SERVER_DB_CONNECTION"
+    )]
+    db_connection: String,
+    /// Logging level of the application
+    #[arg(long = "log-level", default_value_t = log::LevelFilter::Info)]
+    log_level:     log::LevelFilter,
+}
+
+#[derive(Clone)]
+struct AppState {
+    db_connection: Config,
+}
+
+async fn upsert_account_device(
+    Path(device): Path<String>,
+    State(state): State<Arc<AppState>>,
+    Json(account): Json<Vec<String>>,
+) -> Result<impl axum::response::IntoResponse, axum::response::Response> {
+    info!("Subscribing accounts {:?} to device {}", account, device);
+    let _ = &state.db_connection;
+
+    Ok(StatusCode::OK)
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    dotenv().ok();
+    let args = Args::parse();
+
+    tracing_subscriber::fmt::init();
+
+    let db_connection: Config = args.db_connection.parse()?;
+    let app_state = Arc::new(AppState { db_connection });
+
+    let app = Router::new()
+        .route(
+            "/api/v1/device/:device/subscriptions",
```

Comment:
```suggestion
            "/api/v1/device/:device/subscribe",
```

`subscriptions` in my opinion would be a good name for a `GET` endpoint while `subscribe` might refer to that an action `PUT` is done.

Reply:
In general the path should not reflect commands but resources. 

subscribe sounds like a command, whereas subscriptions would be the results of a subscribe action.

Reply:
`/api/v1/device/:device/subscription` and a let a subscription consist of multiple account?

Reply:
ok, then let's leave it for now.